onnxruntime/core/platform/windows/env.cc

```diff
@@ -124,52 +147,171 @@ class WindowsEnv : public Env {
   }
 
   int GetNumCpuCores() const override {
-    SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
-    DWORD returnLength = sizeof(buffer);
-    if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
-      // try GetSystemInfo
-      SYSTEM_INFO sysInfo;
-      GetSystemInfo(&sysInfo);
-      if (sysInfo.dwNumberOfProcessors <= 0) {
-        ORT_THROW("Fatal error: 0 count processors from GetSystemInfo");
+    int num_cores = 0;
+    for (const auto& group_info : group_info_vec_) {
+      num_cores += group_info.ActiveProcessorCount >> 1;
```

Comment:
> num_cores += group_info.ActiveProcessorCount >> 1;

Can you please explain this right shift?

Reply:
ActiveProcessorCount stands for logic processor, onnxruntime need 1 thread per 2 logic processors.

Reply:
Why is it 1 thread per 2 logic processors and not 1 or 4?